x-pack/plugins/ml/public/application/jobs/new_job/common/job_creator/categorization_job_creator.ts

```diff
@@ -75,6 +76,11 @@ export class CategorizationJobCreator extends JobCreator {
   private _createDetector(agg: Aggregation, field: Field) {
     const dtr: Detector = createBasicDetector(agg, field);
     dtr.by_field_name = mlCategory.id;
+
+    // API requires if per_partition_categorization is enabled, add partition field to the detector
+    if (this.perPartitionCategorization && this.categorizationPerPartitionField !== null) {
```

Comment:
If per-partition is enabled inside the categorization job wizard, we should automatically add the selected partition field to the list of `influencers` in the `analysis_config` along with `mlcategory`. If they are in the advanced wizard, leave it up to the user.

Reply:
when changing partition field, the previous one should be removed from the influencer list.
so the list only ever contains `mlcategory` and the selected partition field.

Reply:
Updated here https://github.com/elastic/kibana/pull/75061/commits/7d8c39b0fc8d740f237a4fe9f3c4ef6644c5a4ec